cpp/examples/build.sh

```diff
@@ -25,6 +42,10 @@ build_example() {
   cmake -S ${example_dir} -B ${build_dir} -Dcudf_ROOT="${LIB_BUILD_DIR}"
   # Build
   cmake --build ${build_dir} -j${PARALLEL_LEVEL}
+  # Install if needed
+  if [ "$INSTALL_EXAMPLES" = true ]; then
+    cmake --install ${build_dir} --prefix ${INSTALL_PREFIX:-${PREFIX:-${CONDA_PREFIX:-${example_dir}/install}}}
```

Comment:
Where does `INSTALL_PREFIX` come from? Do we need all of these environment variables? Could we separate this onto another line?

I'd propose something like:
```suggestion
    if [ "$CONDA_BUILD" == "1" ]; then
      INSTALL_PREFIX="${INSTALL_PREFIX:-${PREFIX}}"
    fi
    INSTALL_PREFIX="${INSTALL_PREFIX:-${example_dir}/install}"
    cmake --install ${build_dir} --prefix ${INSTALL_PREFIX}
```

Reply:
We don't need all these, just any one of them (in the preference order) set in the environment. I am partial to removing `$CONDA_PREFIX` from this chain since it adds a possibly unrelated path. Afaik `$INSTALL_PREFIX` is not used by any of the automated tests so it allows users to override the prefix, whereas its first fallback on `$PREFIX` is used by CI to correctly run stuff.

Reply:
Afaik `$INSTALL_PREFIX` is not being used by CI so it can be used by user to provide a superseding path. 

Only of the variables in the chain need to be set.

If `$INSTALL_PREFIX` isn't set, we should set to `${PREFIX}` if running from CI (is `$CONDA_BUILD="1"` check good for this?), else to the local `${example_dir}/install` like you suggested.

Reply:
> Small question, is the `CONDA_BUILD` set to `1` by the CI build? I couldn't find this env var in the recent CI job.

Yes. It's set by `conda-build`.

Reference: https://docs.conda.io/projects/conda-build/en/stable/user-guide/environment-variables.html#environment-variables-set-during-the-build-process

Reply:
> Afaik `$INSTALL_PREFIX` is not being used by CI so it can be used by user to provide a superseding path.
> 
> Only of the variables in the chain need to be set.
> 
> If `$INSTALL_PREFIX` isn't set, we should set to `${PREFIX}` if running from CI (is `$CONDA_BUILD="1"` check good for this?), else to the local `${example_dir}/install` like you suggested.

Yes, you're reading my intentions correctly! I hoped it would be self-explanatory, and you got it right. 👍 Let's apply the suggestion above.

Reply:
Perfect. Let me do that!